Documentation/RepoCompose.md

```diff
@@ -0,0 +1,289 @@
+
+``` json
+"prereq": { 
+    "Microsoft Visual Studio": "2015",
+    "CMake" : "1.0",
+}
+```
+
+A full sample output for `consumes` is available in the Samples section.
+
+### produces
+
+The `produces` command returns json output which describes the artifacts produced by the the repo.  This includes NuGet packages and file artifacts.  
+
+The output format for artifacts is special for `produces` because it lacks any hard location information.  For example: 
+
+- NuGet artifacts lack feeds
+- File artifacts lack a `"kind"` and supporting values
+
+This is because the `produces` command represents "what" a repo produces, not "where" the repo produces it.  The "where" portion is controled by the `publish` command.  External components will take the output of `produces`, add the location information in and feed it back to `publish`.  
+
+Like `consumes` the `produces` output is also grouped by the operating system:
+
+``` json
+{
+    "os-windows": {
+        "nuget": { },
+        "file": { }
+    },
+    "os-linux": {
+        "nuget": { },
+        "file": { }
+    }
+}
+```
+
+A ful sample output for `produces` is available in the Samples section.
+
+### change
+
+The `change` command is used to alter the floating build dependencies section.  It can establish new versions of NuGet packages, new locations to find file artifacts, different NuGet feeds, etc ...  
+
+This is the command which allow us to use the build output of one repo as the input of a dependent repo.  The first repo can build using a new output version (say beta5) and dependent repos can be changed to accept this new version.  
+
+This command operates by providing json whose format is a subset of the output of `consumes`.  In particular it will provide the `"dependencies.floating"` section.  
+
+``` json
+"dependencies" {
+    "floating": { 
+        "nuget": {
+            "packages" {
+                "MicroBuild.Core": "0.2.0",
+                "Microsoft.NETCore.Platforms": "1.0.1"
+            }
+        }
+    }
+}
+```
+
+A tool responsible for composing repos would ideally:
+
+1. Execute `run.cmd consumes` and capture the output.
+2. Alter the NuGet package versions in the json to have the correct build identifier: beta5, RTM, etc ..
+3. Execute `run.cmd change` and pass in the altered output.
+
+
+### publish
+
+The `publish` command takes a json input that describes the locations artifacts should be published to.  The input to this command is the output of `produces` that is augmented with location information.  
+
+## Artifact Specification
+
+The json describing artifacts is the same between the `consume`, `produces` and `publish` commands.  These items can be used anywhere artifacts are listed above.
+
+### NuGet packages
+
+The description for NuGet artifacts is it two parts:
+
+1. The set of feeds packages are being read from.
+2. The set of packages that are being consumed and their respective versions.
```

Comment:
I hope this doesn't lead to information rot as feeds change and people unfamiliar with the specifics don't know what to do with the feed list, you end up with unused feeds listed.  It's more of a problem / responsibility for the repo owner, but I could see it happening.